lib/ansible/modules/mount_facts.py

```diff
@@ -0,0 +1,650 @@
+
+    Each tuple contains the mount point, line of origin, and the dictionary of the parsed line.
+    """
+    if not any(len(fields[4]) == 10 for line in lines for fields in [line.split()]):
+        raise ValueError
+    for line in lines:
+        fields = line.split()
+        datetime.date.fromtimestamp(int(fields[4]))
+        mount_info: dict[str, str | int] = {
+            "device": fields[0],
+            "mount": fields[1],
+            "fstype": fields[2],
+            "options": fields[3],
+            "time": int(fields[4]),
+        }
+        yield MountInfo(fields[1], line, mount_info)
+
+
+def gen_mounts_by_file(file: str) -> t.Iterable[MountInfo | MountInfoOptions]:
+    """Yield parsed mount entries from the first successful generator.
+
+    Generators are tried in the following order to minimize false positives:
+    - /etc/vfstab: 7 columns
+    - /etc/mnttab: 5 columns (mnttab[4] must contain UNIX timestamp)
+    - /etc/fstab: 4-6 columns (fstab[4] is optional and historically 0-9, but can be any int)
+    - /etc/filesystems: multi-line, not column-based, and specific to AIX
+    """
+    if (lines := get_file_content(file, "").splitlines()):
+        for gen_mounts in [gen_vfstab_entries, gen_mnttab_entries, gen_fstab_entries, gen_aix_filesystems_entries]:
+            with suppress(IndexError, ValueError):
+                # mpypy error: misc: Incompatible types in "yield from" (actual type "object", expected type "Union[MountInfo, MountInfoOptions]
+                # only works if either
+                # * the list of functions excludes gen_aix_filesystems_entries
+                # * the list of functions only contains gen_aix_filesystems_entries
+                yield from list(gen_mounts(lines))  # type: ignore[misc]
+                break
+
+
+def get_sources(module: AnsibleModule) -> list[str]:
+    """Return a list of filenames from the requested sources."""
+    sources: list[str] = []
+    for source in module.params["sources"] or ["all"]:
+        if not source:
+            module.fail_json(msg="sources contains an empty string")
+
+        if source in {"dynamic", "all"}:
+            sources.extend(DYNAMIC_SOURCES)
+        if source in {"static", "all"}:
+            sources.extend(STATIC_SOURCES)
+
+        elif source not in {"static", "dynamic", "all"}:
+            sources.append(source)
+    return sources
+
+
+def gen_mounts_by_source(module: AnsibleModule):
+    """Iterate over the sources and yield tuples containing the source, mount point, source line(s), and the parsed result."""
+    sources = get_sources(module)
+
+    if len(set(sources)) < len(sources):
+        module.warn(f"mount_facts option 'sources' contains duplicate entries, repeat sources will be ignored: {sources}")
+
+    mount_fallback = module.params["mount_binary"] and set(sources).intersection(DYNAMIC_SOURCES)
+
+    seen = set()
+    for source in sources:
+        if (real_source := os.path.realpath(source)) in seen:
```

Comment:
This does have the side-effect of turning `mount` into a path even though it's not. For the purposes of this check it should be fine, unless one of the actual sources happens to be a `mount` file in the current directory.

Reply:
Oh, good catch. let me fix that